crates/js-sys/src/lib.rs

```diff
@@ -3522,6 +3522,22 @@ impl JsString {
             None
         }
     }
+
+    /// Returns whether this string is a valid UTF-16 string.
```

Comment:
These docs are pretty weird: they mention `as_string`, but `as_string` doesn't exist on `JsString`. Instead, the relevant function is `Into::into`.

I also think it should use "unpaired surrogate" rather than "lone surrogate", since that seems to be the normal term for it (and it also gives better Google results).

Reply:
I think that `JsString` derefs to `Object` which derefs to `JsValue` so the `as_string` method should be inherently available? As for changing terms 👍

Reply:
Sure, that may be true, but [it doesn't show up in the docs](https://docs.rs/js-sys/0.3.17/js_sys/struct.JsString.html), and in addition it's not idiomatic: users should use `foo.into()` to convert from `JsString` to `String`, not `foo.as_string().unwrap_throw()`

Reply:
Ah good point, I'll update!